api-common/src/main/kotlin/io/qalipsis/api/meters/Rate.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 AERIS IT Solutions GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package io.qalipsis.api.meters
+
+/**
+ * Measures the relationship between two independently tracked [Gauge] metrics.
+ *
+ * @author Francisca Eze
+ */
+interface Rate : Meter<Rate> {
+
+    /**
+     * Calculates an instantaneous result gotten from dividing two gauge meters.
+     */
+    fun current(): Double {
+        return Double.NaN
+    }
+
+    /**
+     * Decrease the value of the primary gauge by the `amount`.
+     *
+     * @param amount amount to subtract from the gauge value
+     */
+    fun decrementPrimaryMetric(amount: Double = 1.0): Double
```

Comment:
Do you think that primary and secondary is clear for the users? It is not for me. What about decrementTotal and decrementShare instead?